File: Easy/python/TwoSum.py
def twoSum(nums, target):
    required = {}
    for i in range(len(nums)):
        result = target - nums[i]
        if result in required:
            return [required[result],i]
        else:
            required[nums[i]]=i

if __name__ == '__main__':
    nums = [2,7,11,15]
    target = 9

    result = twoSum(nums,target)
    print(result)

File: Easy/cpp/twoSum.cpp
#include <map>
#include <vector>
#include <iostream>

using namespace std;

class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            map<int, int> hash;

            for (int i = 0; i < nums.size(); i++) 
            {
            int numberToFind = target - nums[i];

            if (hash.find(numberToFind) != hash.end()) 
                return  {hash[numberToFind] ,i };

                hash[nums[i]] = i;
            }
            return {};
        }
        
};
